Userland/Libraries/LibJS/Parser.cpp

```diff
@@ -792,6 +794,8 @@ RefPtr<Statement> Parser::try_parse_labelled_statement(AllowLabelledFunction all
         m_state.current_scope_pusher->add_declaration(function_declaration);
         if (function_declaration->kind() == FunctionKind::Generator)
             syntax_error("Generator functions cannot be defined in labelled statements");
+        if (function_declaration->kind() == FunctionKind::Async)
+            syntax_error("Async functions cannot be defined in labelled statements");
```

Comment:
This can probably just be `kind() != FunctionKind::Regular`. Although that does make the error slightly less clear.

Reply:
I prefer the more detailed error message